pkg/providers/vsphere/vsphere.go

```diff
@@ -431,6 +437,30 @@ func (p *vsphereProvider) SetupAndValidateCreateCluster(ctx context.Context, clu
 		logger.Info("Skipping check for whether control plane ip is in use")
 	}
 
+	vuc := NewVsphereUserConfig()
+
+	if err := p.validator.validateUserPrivs(ctx, vSphereClusterSpec, vuc); err != nil {
+		return err
+	} else {
+		logger.MarkPass("EKSA_VSPHERE_USER vSphere privileges validated")
+	}
+
+	if len(vuc.EksaVsphereCPUsername) > 0 {
```

Comment:
Verify that CP/CSI values exist *and* usernames are different from `EKSA_VSPHERE_USERNAME`.